rest/src/main/java/com/com/rest/endpoint/ProductEndpoint.java

```diff
@@ -39,25 +41,33 @@
     }
 
     @GetMapping("/{id}")
-    public ResponseEntity<Product> getProductById(@PathVariable("id") int id) throws ProductNotFoundException {
+    public ResponseEntity<Product> getProductById(@PathVariable("id") int id, @AuthenticationPrincipal
+                                    CurrentUser currentUser) throws ProductNotFoundException {
+        log.info("User {} searching product by id ", currentUser.getUser().getEmail());
```

Comment:
orinak stex karajarkem  uneneal Security.class mej method vore mezi kuta  current user-in u estex et mase kancheq ete dez anhrajesht e es mase log enel